ui/src/components/Dashboard.tsx

```
@@ -109,151 +126,92 @@ const Dashboard: React.FC = () => {
-
-            const result = await createNewGame(randomSeed);
-
-            if (result.success && result.gameAddress) {
-                setNewGameAddress(result.gameAddress);
+            const gameAddress = selectedContractAddress || DEFAULT_GAME_CONTRACT;
+            
+            // Create the new game using the client's newHand method
+            // The account's nonce is managed automatically by the NodeRpcClient
+            const result = await client.newHand(gameAddress, randomSeed);
+            
+            if (result && result.hash) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)